### PR TITLE
Reader: update header for P2 postcards to show author

### LINKED ISSUE #14875: Update header for P2 postcards to show author
After receiving some feedback internally that it's much more efficient to have the author name included in the postcard UI for P2 posts specifically, I've updated the design for P2 postcards as per below.

Changes to header include:

- Show author name next to site name
- Show author icon next to site icon
- Show timestamp below

Screen Shot 2020-09-10 at 2 56 16 PM

Screen Shot 2020-09-10 at 2 56 19 PM

Specs coming tomorrow (Sep 11)

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #14875 

This adds author information to the Reader card for new P2 sites.

To test:
- Go to the Reader.
  - Tip: The updated P2 card will show in any filter, they're just easier to find in Automattic.
- For _new_ P2 posts, the card will now have the author avatar and name in the header.

| Before | After |
|--------|-------|
| before | after |

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderCellConfiguration.swift [hunk 1]
```diff
@@ -41,13 +41,16 @@ final class ReaderCellConfiguration {
             return
         }
 
-        let postCell = cell as! ReaderPostCardCell
+        guard let postCell = cell as? ReaderPostCardCell else {
+            return
+        }
```
Comment (mindgraffiti) [inline WordPress/Classes/ViewRelated/Reader/ReaderCellConfiguration.swift:47]: much better :)

File: WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift [hunk 2]
```diff
@@ -278,51 +298,94 @@ protocol ReaderTopicsChipsDelegate: class {
...
+            Constants.rotate270Degrees
+
+        arrowImageView.transform = CGAffineTransform(rotationAngle: imageRotationAngle)
+    }
+
+    private func configureAvatarImageView(_ imageView: UIImageView) {
+        imageView.layer.borderColor = WPStyleGuide.readerCardBlogIconBorderColor().cgColor
+        imageView.layer.borderWidth = Constants.imageBorderWidth
+        imageView.layer.masksToBounds = true
     }
```
Comment (mindgraffiti) [inline WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift:389]: The ReaderCrossPostCell avatars could benefit from this treatment too. wdyt?

Simulator Screen Shot - iPhone 11 Pro - 2020-11-17 at 14 32 29

**Review by mindgraffiti — APPROVED** [review]
🔢  1 suggestion, no blockers
✅  UI tested

:shipit:

File: WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift (hunk 2 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Reader/ReaderPostCardCell.swift:389]: Sure. I'll look at it separately though if that's ok.

**Pull request merged**